packages/angular/src/utils/nx-devkit/ast-utils.ts

```diff
@@ -676,6 +676,24 @@ function getListOfRoutes(
   return null;
 }
 
+export function isNgStandaloneApp(tree: Tree, projectName: string) {
+  ensureTypescript();
+  const { tsquery } = require('@phenomnomnominal/tsquery');
+
+  const project = getProjects(tree).get(projectName);
+  const mainFile = project.targets.build.options.main;
```

Comment:
I understand the helper is meant to be invoked only for apps, so the consumer must ensure this is invoked for an app, but there's no proper error here to provide feedback when that's not the case.

We could silently handle it and return `false`, or we could throw an error, but I'd just return `false` for simplicity and nothing really breaks as long as it's handled here:

```suggestion
  const project = readProjectConfiguration(tree, projectName);
  const mainFile = project.targets?.build?.options?.main;
  
  if (project.type !== 'application' || !mainFile) {
    return false;
  }
  
  ensureTypescript();
  const { tsquery } = require('@phenomnomnominal/tsquery');
```